android/app/src/main/java/app/organicmaps/search/SearchFragment.java

```diff
@@ -240,6 +244,31 @@ private void updateResultsPlaceholder()
     UiUtils.showIf(show, mResultsPlaceholder);
   }
 
+  private void updateViewsInsets()
+  {
+    ViewCompat.setOnApplyWindowInsetsListener(requireView(), (view, windowInsets) -> {
+      final Insets insets = windowInsets.getInsets(WindowInsetUtils.TYPE_SAFE_DRAWING);
+      int navBarHeight = insets.bottom;
+
+      int baseMargin = UiUtils.dimen(view.getContext(), R.dimen.margin_base);
+
+      MarginLayoutParams buttonLayoutParams = (MarginLayoutParams) mShowOnMapFab.getLayoutParams();
+      int buttonBottomMargin = navBarHeight + baseMargin;
+      buttonLayoutParams.bottomMargin = buttonBottomMargin;
+
+      RecyclerView resultsRecycler = mResultsFrame.findViewById(R.id.recycler);
+
+      int recyclerPaddingBottom = buttonBottomMargin
+                                  + mShowOnMapFab.getMeasuredHeight()
+                                  + baseMargin;
+      resultsRecycler.setPadding(
+          insets.left, resultsRecycler.getPaddingTop(),
+          insets.right, recyclerPaddingBottom);
+
+      return windowInsets;
+    });
+  }
```

Comment:
Can we avoid copy-paste with BookmarksListFragment? Take out into some base fragment class or InsetUtils?

Reply:
Done. `ScrollableContentInsetsListener` now supports both regular lists and considers floating action button when provided